test/tests/mendeleyImport.js

```diff
@@ -1,12 +1,34 @@
-/* global setHTTPResponse:false, sinon: false, Zotero_Import_Mendeley: false */
+/* global setHTTPResponse:false, sinon: false, Zotero_Import_Mendeley: false, HttpServer: false */
 
 describe('Zotero_Import_Mendeley', function () {
-	var server, importer;
+	var server, importer, httpd, httpdURL;
 
 	before(async () => {
 		Components.utils.import('chrome://zotero/content/import/mendeley/mendeleyImport.js');
 		importer = new Zotero_Import_Mendeley();
 		importer.mendeleyCode = 'CODE';
+
+		// real http server is used to deliver an empty pdf so that annotations can be processed during import
+		Components.utils.import("resource://zotero-unit/httpd.js");
+		const emptyPDF = await Zotero.File.getContentsFromURLAsync('resource://zotero-unit-tests/data/empty.pdf');
+		const port = 16213;
+		httpd = new HttpServer();
+		httpdURL = `http://127.0.0.1:${port}`;
+		httpd.start(port);
+		httpd.registerPathHandler('/file1.pdf',
+			{
+				handle: function (request, response) {
+					response.setStatusLine(null, 200, "OK");
+					response.setHeader('Content-Type', 'application/json', false);
```

Comment:
Is this supposed to be `application/json`?

If you just want to serve the file, you can probably just use `httpd.registerFile()` ([example](https://github.com/zotero/zotero/blob/3b993b2af346ce8c1d37282808f9b3ab2d16c5a0/test/tests/attachmentsTest.js#L794-L797)).

Reply:
Thanks for point to the example, that's much nicer. (also 100% shouldn't have been `application/json`)